### PR TITLE
Add query to check if userdata contains private key. Closes #2274

### LINKED ISSUE #2274: Add "User Data Base64 Contains Encoded Private Key" query for CloudFormation
### Platform
CloudFormation

### Description
Add support in CloudFormation for the already implemented query in Terraform.
User Data Base64 contains an encoded RSA Private Key.

### PR DESCRIPTION
**Pull request by joaomartinscx**
Closes #2274

**Proposed Changes**

- Add support in CloudFormation for the query in Terraform. It checks if the user data in the launch configuration, encoded in base64, contains an RSA private key.
- Changed the query names to be accordingly to the unified name

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/ansible/aws/user_data_contains_rsa_private_key/metadata.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (Ruben-Silva) [inline assets/queries/ansible/aws/user_data_contains_rsa_private_key/metadata.json:1]: Change folder name to match query_name?

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/metadata.json:1]: Change folder name to match query_name?

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+Resources:
+  myLaunchConfig:
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml:3]: Change "myLaunchConfig" to "myLaunchConfig2"

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "Resources":{
+    "myLaunchConfig":{
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.json:3]: Change "myLaunchConfig" to "myLaunchConfig3"

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+Resources:
+  myLaunchConfig:
```
Comment (Ruben-Silva) [inline assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.yaml:3]: Change "myLaunchConfig" to "myLaunchConfig4"

File: assets/queries/terraform/aws/user_data_base64_contains_encoded_private_key/metadata.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (Ruben-Silva) [inline assets/queries/terraform/aws/user_data_base64_contains_encoded_private_key/metadata.json:1]: Change folder name to match query_name?

**Review by Ruben-Silva — CHANGES_REQUESTED** [review]

File: assets/queries/ansible/aws/user_data_contains_rsa_private_key/metadata.json (hunk 1 above)
Comment (joaomartinscx) [reply assets/queries/ansible/aws/user_data_contains_rsa_private_key/metadata.json:1]: done

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml (hunk 3 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml]: done

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/metadata.json (hunk 2 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/metadata.json]: done

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml (hunk 3 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml]: Why rename for different samples?

File: assets/queries/terraform/aws/user_data_base64_contains_encoded_private_key/metadata.json (hunk 6 above)
Comment (joaomartinscx) [reply assets/queries/terraform/aws/user_data_base64_contains_encoded_private_key/metadata.json:1]: done

**Review by ruigomescx — DISMISSED** [review]

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml (hunk 3 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/negative.yaml]: done

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.json (hunk 4 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.json]: done

File: assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.yaml (hunk 5 above)
Comment (joaomartinscx) [reply assets/queries/cloudFormation/user_data_base64_contains_encoded_private_key/test/positive.yaml]: done

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**